app/scripts/migrations/037.js

```diff
@@ -0,0 +1,47 @@
+// next version number
+const version = 37
+const { getFinalStates, containsFinalStates } = require('../controllers/transactions/lib/util')
+const { DEFAULT_TX_HISTORY_LIMIT } = require('../controllers/transactions/enums')
+
+/*
+
+
+*/
+
+const clone = require('clone')
+
+module.exports = {
+  version,
+
+  migrate: async function (originalVersionedData) {
+    const versionedData = clone(originalVersionedData)
+    versionedData.meta.version = version
+    const state = versionedData.data
+    const newState = transformState(state)
+    versionedData.data = newState
+    return versionedData
+  },
+}
+
+function transformState (state) {
+  const newState = state
+  // transform state here
+  if (newState.TransactionController) {
+    if (newState.TransactionController.transactions) {
+      const transactions = newState.TransactionController.transactions
+      let txCount = transactions.length
+      if (txCount > DEFAULT_TX_HISTORY_LIMIT) {
+        do {
+          const index = transactions.findIndex((txMeta) => {
+            return getFinalStates().includes(txMeta.status)
```

Comment:
Similarly here, should we cache the final states array instead of creating a new one per tx?

Reply:
that function wont return a new array but i dont want to remove the function and just replace it with the array